docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -0,0 +1,378 @@
+        signer = header.signer
+
+        try:
+            game_name, action, space = self._decode_data(transaction.payload)
+        except:
+            raise InvalidTransaction("Invalid payload serialization")
+
+        return signer, game_name, action, space
+
+{% endif %}
+
+
+Before we say how exactly the transaction payload will be decoded, let's look
+at ``_get_state_data``. Now, as far as the handler is concerned, it doesn't
+matter how the game data is stored. The only thing that matters is that given a
+game name, the state store is able to give back the correct game data. (In our
+full XO implementation, the game data is stored in a Merkle-radix tree.)
+
+
+{% if language == 'JavaScript' %}
+
+.. TODO:: 
+
+    Example code to be provided by JavaScript expert, along with 
+    rewrite suggestion for surrounding text.
+
+{% else %}
+
+.. code-block:: python
+
+    def _get_state_data(self, game_name, state_store):
+        game_address = self._make_game_address(game_name)
+
+        state_entries = state_store.get([game_address])
+
+        try:
+            return self._decode_data(state_entries[0].data)
+        except IndexError:
+            return None, None, None, None
+        except:
+            raise InternalError("Failed to deserialize game data.")
+
+
+{% endif %}
+
+It doesn't matter what exactly the game address is. By convention, we'll store
+game data at an address obtained from hashing the game name prepended with some
+constant:
+
+{% if language == 'JavaScript' %}
+
+.. TODO::
+
+    Example code to be provided by JavaScript expert, along with 
+    rewrite suggestion for surrounding text.
+    
+{% else %}
+
+.. code-block:: python
+
+    def _make_game_address(self, game_name):
+        prefix = self._namespace_prefix
+        game_name_utf8 = game_name.encode('utf-8')
+        return prefix + hashlib.sha512(game_name_utf8).hexdigest()
+
+
+{% endif %}
+
+Finally, we'll store the game data. To do this, we simply need to encode the
+updated state of the game and store it back at the address from which it came.
+
+{% if language == 'JavaScript' %}
+
+.. TODO::
+
+    Example code to be provided by JavaScript expert, along with 
+    rewrite suggestion for surrounding text.
+
+{% else %}
+
+.. code-block:: python
+
+    def _store_game_data(self, game_name, game_data, state_store):
+        game_address = self._make_game_address(game_name)
+
+        encoded_game_data = self._encode_data(game_data)
+
+        addresses = state_store.set([
+            StateEntry(
+                address=game_address,
+                data=encoded_game_data
+            )
+        ])
+
+        if len(addresses) < 1:
+            raise InternalError("State Error")
+
+{% endif %}
+
+So, how should we encode and decode the data? In fact, we can choose whatever
+encoding scheme we want; the data is only going to get read and written by the
+handler, so as long as we're consistent, it doesn't matter. In this case, we'll
```

Comment:
Instead of saying 'consistent' and 'it doesn't matter', I would say "...we have many options in binary encoding schemes; the binary data stored in the validator state is up to the implementer of the handler.